src/index.ts

```diff
@@ -1841,6 +1880,28 @@ export default class Decimal {
     return this.recip();
   }
 
+  //Taken from OmegaNum.js, with a couple touch-ups
+  public mod(value: DecimalSource): Decimal {
```

Comment:
it might be worth documenting which kind of modulo this is intended to be: https://en.wikipedia.org/wiki/Modulo#In_programming_languages

Looks like it's Euclidean whereas Javascript's is Truncated.

Reply:
Looks like it has some precision problems even when both numbers are layer 0:

new Decimal("1e14").mod(1.567).toString()
'0.875'
1e14 % 1.567
0.8812874504366692

If both numbers can be expressed as Javascript doubles, then you could turn to doubles, mod and return the result (after turning it from Truncated to Euclidean of course).

Reply:
Found a bug in the handling of mod when the 'this is too small compared to decimal' case happens, you don't do the euclidean modulo's requirement for sign multiplication:

new Decimal("1e-10000").mod("2").toString()
'1e-10000'
new Decimal("1e-10000").mod("-2").toString()
'1e-10000'

Reply:
Ah, I thought this matched JavaScript's - I failed to consider that the behavior could be different when the second number is negative. I'm willing to change it - would you rather have modulo match JavaScript's version or OmegaNum's version?

Reply:
Hard to say - my preference is actually Floored, which matches neither of them. Making it match the Javascript behaviour would make sense, but it's not like break_eternity is specifically a Javascript library, it just happens that this implementation is in Javascript.

To try to make a more objective decision, I looked at languages I think it's more likely this language would be ported to and tallied up how they implement %/mod:

Truncated: C/C++, C#, Go, Java, JavaScript, Rust, SQL, WebAssembly, GDScript
Euclidean:
Floored: Lua, Python, Ruby

So I think Truncated is the 'objective' correct choice, though you could do Floored if you're feeling spicy.